### PR TITLE
DocC setup

### PR DESCRIPTION
**Pull request by yim-lee**

### CODE REVIEW

**Comment by yim-lee** [comment]
Need https://github.com/apple/swift-service-discovery/pull/41 for 5.6 and 5.7 builds.

File: scripts/soundness.sh [hunk 1]
```diff
@@ -60,7 +60,7 @@ for language in swift-or-c bash dtrace; do
   matching_files=( -name '*' )
   case "$language" in
       swift-or-c)
-        exceptions=( -name c_nio_http_parser.c -o -name c_nio_http_parser.h -o -name cpp_magic.h -o -name Package.swift -o -name CNIOSHA1.h -o -name c_nio_sha1.c -o -name ifaddrs-android.c -o -name ifaddrs-android.h)
+        exceptions=( -name c_nio_http_parser.c -o -name c_nio_http_parser.h -o -name cpp_magic.h -o -name Package.swift -o -name Package@swift-5.6.swift -o -name CNIOSHA1.h -o -name c_nio_sha1.c -o -name ifaddrs-android.c -o -name ifaddrs-android.h)
```
Comment (tomerd) [inline scripts/soundness.sh:63]: `Package@swift-5.6.swift` -> `Package@*.swift`

Comment (tomerd) [reply scripts/soundness.sh:63]: also not sure we need al this nio files, maybe copy pasta?

File: Sources/ServiceDiscovery/Documentation.docc/index.md [hunk 2]
```diff
@@ -0,0 +1,193 @@
+# ``ServiceDiscovery``
```
Comment (tomerd) [inline Sources/ServiceDiscovery/Documentation.docc/index.md:1]: https://github.com/apple/swift-statsd-client/pull/36/files#r934924113

**Review by ktoso — APPROVED** [review]
We should wrap over the Package.swift style to be for the "old versions", but otherwise LGTM~!

**Comment by tomerd** [comment]
> We should wrap over the Package.swift style to be for the "old versions", but otherwise LGTM~!

I mentioned this on the other PR, this is something @fabianfett and myself tried with aws-lambda-runtime and had trouble with. I cant remember all the details but @fabianfett had issues making the "old versions" work reliably and we ended up adding a 5.6 version as the PR does currently.

**Comment by ktoso** [comment]
I guess let's figure it out over here and then apply to all repos: https://github.com/apple/swift-statsd-client/pull/36 
Commented over there a bit

File: Package@swift-5.4.swift [hunk 3]
```diff
@@ -0,0 +1,36 @@
+// swift-tools-version:5.3
```
Comment (tomerd) [inline Package@swift-5.4.swift:1]: 5.4?

File: Package@swift-5.5.swift [hunk 4]
```diff
@@ -0,0 +1,36 @@
+// swift-tools-version:5.3
```
Comment (tomerd) [inline Package@swift-5.5.swift:1]: 5.5?

**Review by tomerd — APPROVED** [review]
nice <3

File: scripts/preview_docc.sh [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+## Licensed under Apache License v2.0
+##
+## See LICENSE.txt for license information
+## See CONTRIBUTORS.md for the list of Swift Distributed Actors project authors
+##
+## SPDX-License-Identifier: Apache-2.0
+##
+##===----------------------------------------------------------------------===##
+
+xcrun swift package --disable-sandbox preview-documentation --target $1
```
Comment (tomerd) [inline scripts/preview_docc.sh:31]: I dont think `xcrun` is required, and it is not available on linux. what was the reason of using it here?

cc @ktoso

**Review by tomerd — APPROVED** [review]

File: Package@swift-5.4.swift (hunk 3 above)
Comment (tomerd) [inline Package@swift-5.4.swift:1]: 5.4?

File: scripts/preview_docc.sh [hunk 6]
```diff
@@ -0,0 +1,31 @@
+#!/bin/bash
+
+##===----------------------------------------------------------------------===##
+##
+## This source file is part of the Swift Service Discovery open source project
+##
+## Copyright (c) 2022 the SwiftStatsdClient project authors
```
Comment (tomerd) [inline scripts/preview_docc.sh:7]: ServiceDiscovery? should the soundness script catch this?

Comment (yim-lee) [reply scripts/preview_docc.sh]: Yes, there was an issue with the script but it didn't fail:

```
bash... find: Package@swift-5.1.swift: unknown primary or operator
```

Fixed in https://github.com/apple/swift-service-discovery/pull/42/commits/c5e6f560a5da4006ca56c9468f42e3710a66aa5d

**Pull request merged**